### PR TITLE
Adds a hidden testing page

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

This has useful functions for developers and testers but should be hidden
from production users. The functions for now are: debug, purge, wipe all data.

medic/cht-core#5935

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/js/controllers/about.js [hunk 1]
```diff
@@ -53,13 +51,22 @@ angular.module('inboxControllers').controller('AboutCtrl',
       })
       .then(rev => ctrl.remoteRev = rev);
 
+    ctrl.knockCount = 0;
+    ctrl.secretDoor = () => {
+      if (ctrl.doorTimeout) {
+        $timeout.cancel(ctrl.doorTimeout);
+      }
+      if (ctrl.knockCount++ >= 5) {
+        $state.go('testing');
```
Comment (kennsippell) [inline webapp/src/js/controllers/about.js]: just wondering if there is a way back to the main app after entering the testing state.  it requires browser back button? i didn't actually run this code

File: webapp/src/js/controllers/testing.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        });
+      }
+      return $q.resolve();
+    };
+
+    const wipeDatabases = () => {
+      return $q
+        .all([
+          DB({ remote: false }).destroy(),
+          DB({ remote: false, meta: false }).destroy()
```
Comment (kennsippell) [inline webapp/src/js/controllers/testing.js]: is the default `meta: false`? is this destroying the meta database twice?

File: webapp/src/js/controllers/testing.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        ])
+        .catch(err => $log.error('Error wiping databases', err));
+    };
+
+    const wipeCookies = () => {
+      [ 'locale', 'userCtx', 'medic-webapp-debug' ].forEach(name => {
```
Comment (kennsippell) [inline webapp/src/js/controllers/testing.js]: consider clearing all cookies via `ipCookie.get()` so we don't forget to maintain this list

File: webapp/src/js/controllers/testing.js (hunk 2 above)
Comment (garethbowen) [reply webapp/src/js/controllers/testing.js]: :woman_facepalming: Fixed...

File: webapp/src/js/controllers/about.js (hunk 1 above)
Comment (garethbowen) [reply webapp/src/js/controllers/about.js]: You get all the usual navigation buttons you had before so you can easily switch to another tab:

Screenshot from 2019-11-20 09-12-20

The back button also works.

File: webapp/src/js/controllers/testing.js (hunk 3 above)
Comment (garethbowen) [reply webapp/src/js/controllers/testing.js]: Good call. Done.

**Pull request merged**